modules/delegates/api/fetchDelegates.ts

```diff
@@ -302,3 +314,281 @@ export async function fetchDelegates(
+      expirationDate,
+      expired: expirationDate > new Date() ? false : true,
+      isAboutToExpire: isAboutToExpireCheck(expirationDate),
+      name: ghDelegate?.name,
+      tags: ghDelegate?.tags,
+      previous:
+        oldOwner && oldContractAddress
+          ? { address: oldOwner, voteDelegateAddress: oldContractAddress }
+          : undefined,
+      next:
+        newOwner && newContractAddress
+          ? { address: newOwner, voteDelegateAddress: newContractAddress }
+          : undefined
+    };
+  });
+
+  return [githubDelegates, allDelegatesWithNamesAndLinks];
+}
+
+export async function fetchSingleDelegateInfo(
+  address: string,
+  network: SupportedNetworks
+): Promise<DelegateInfo | null> {
+  const [githubDelegates, allDelegatesWithNamesAndLinks] = await fetchAndMergeDelegates(network);
+
+  const foundDelegate = allDelegatesWithNamesAndLinks.find(
+    delegate => delegate.voteDelegate.toLowerCase() === address.toLowerCase()
+  );
+
+  if (!foundDelegate) {
+    return null;
+  }
+
+  const foundGithubDelegate = githubDelegates?.find(delegate => delegate.name === foundDelegate?.name);
+
+  return {
+    name: foundDelegate.name || 'Shadow Delegate',
+    picture: foundGithubDelegate?.picture,
+    address: foundDelegate.delegate,
+    voteDelegateAddress: foundDelegate.voteDelegate,
+    status: foundGithubDelegate ? DelegateStatusEnum.recognized : DelegateStatusEnum.shadow,
+    cuMember: foundGithubDelegate?.cuMember,
+    pollParticipation: foundGithubDelegate?.pollParticipation,
+    executiveParticipation: foundGithubDelegate?.executiveParticipation,
+    combinedParticipation: foundGithubDelegate?.combinedParticipation,
+    communication: foundGithubDelegate?.communication,
+    blockTimestamp: foundDelegate.blockTimestamp,
+    expirationDate: foundDelegate.expirationDate,
+    expired: foundDelegate.expired,
+    isAboutToExpire: foundDelegate.isAboutToExpire,
+    previous: foundDelegate.previous,
+    next: foundDelegate.next
+  };
+}
+
+export async function fetchDelegatesInfo(
+  network: SupportedNetworks,
+  recognizedOnly: boolean,
+  includeExpired: boolean
+): Promise<DelegateInfo[]> {
+  const [githubDelegates, allDelegatesWithNamesAndLinks] = await fetchAndMergeDelegates(network);
+
+  const delegatesInfo = allDelegatesWithNamesAndLinks
+    .filter(
+      delegate =>
+        (recognizedOnly ? delegate.delegateType === DelegateTypeEnum.RECOGNIZED : true) &&
+        (includeExpired ? true : !delegate.expired)
+    )
+    .filter((delegate, i, arr) =>
+      recognizedOnly && !includeExpired ? arr.findIndex(d => d.name === delegate.name) === i : true
+    )
+    .sort((a, b) => new Date(a.blockTimestamp).getTime() - new Date(b.blockTimestamp).getTime())
+    .map(delegate => {
+      const githubDelegate = githubDelegates?.find(d => d.name === delegate.name);
+      return {
+        name: delegate.name || 'Shadow Delegate',
+        picture: githubDelegate?.picture,
```

Comment:
this code merging the delegates structure is repeated 3 times in this page, please, move it to another function and make the other functions reuse it, if possible